static/app/views/monitors/components/monitorCreateForm.tsx

```diff
@@ -82,11 +87,21 @@ function MockTimelineVisualization(props: ScheduleConfig) {
     `/organizations/${organization.slug}/monitors-schedule-data/`,
     {query},
   ] as const;
-  const {data, isLoading} = useApiQuery<number[]>(sampleDataQueryKey, {
+  const {data, isLoading, isError, error} = useApiQuery<number[]>(sampleDataQueryKey, {
     staleTime: 0,
-    enabled: isValidConfig(props),
+    enabled: isValidConfig(schedule),
+    retry: false,
   });
 
+  const errorMessage =
+    isError || !isValidConfig(schedule)
+      ? error?.responseJSON?.schedule?.[0] ?? t('Invalid Schedule')
+      : null;
+
+  useEffect(() => {
+    onError(errorMessage);
+  }, [errorMessage, onError]);
```

Comment:
Have to be careful about how `onError` get's defined here, since if you use this component and have a `onError={(e) => whatever}` it will always change value every render and trigger the error every time :(